src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -79,4 +127,75 @@ public UserTenantCollection getByUsernameAndTenantId(UUID consortiumId, String u
     result.setTotalRecords(1);
     return result;
   }
+
+  private User prepareShadowUser(UUID userId, UserTenant userTenantDto) {
+    User userOptional = getUser(userId);
+    User user = new User();
+    if (Objects.nonNull(userOptional.getId())) {
+      user.setId(UUID.randomUUID().toString());
+      user.setPatronGroup(PATRON_GROUP);
+      user.setUsername(userOptional.getUsername() + HelperUtils.randomString(RANDOM_STRING_COUNT));
+      if(Objects.nonNull(userOptional.getPersonal())) {
+        Personal personal = new Personal();
+        personal.setLastName(userOptional.getPersonal().getLastName());
+        personal.setFirstName(userOptional.getPersonal().getFirstName());
+        personal.setEmail(userOptional.getPersonal().getEmail());
+        personal.setPreferredContactTypeId(userOptional.getPersonal().getPreferredContactTypeId());
+        user.setPersonal(personal);
+      }
+      user.setPatronGroup(userOptional.getPatronGroup());
+      user.setActive(IS_PRIMARY_TRUE);
+    } else {
+      throw new ResourceNotFoundException(USER_ID, userId.toString());
+    }
+    prepareContextForTenant(userTenantDto.getTenantId());
+    return user;
+  }
+
+  private void validateAndUpdateShadowUser(UUID userId) {
+    updateUser(getUser(userId));
+  }
+
+  private User getUser(UUID userId) {
+    log.info("Getting user by userId {}.", userId);
+    return usersClient.getUsersByUserId(String.valueOf(userId));
+  }
+
+  private void updateUser(User user) {
+    if (HelperUtils.existingUserUpToDate(user)) {
+      log.info("{} is up to date.", user);
+    } else {
+      user.setActive(IS_PRIMARY_TRUE);
+      log.info("Updating {}.", user);
+      usersClient.updateUser(user.getId(), user);
+    }
+  }
+
+  private void prepareContextForTenant(String tenantId) {
```

Comment:
When we deleted HelperUtils.getTenantId(folioExecutionContext); -> we don't need this method any more, correct?

Reply:
this method is used for intializing and switching between the folio context for requested tenants .